server/opts.go

```diff
@@ -1842,6 +1846,11 @@ func parseRemoteLeafNodes(v interface{}, errors *[]error, warnings *[]error) ([]
 				}
 			}
 		}
+		if remote.ShuffleURLs {
+			rand.Shuffle(len(remote.URLs), func(i, j int) {
```

Comment:
I would not do the shuffling in parsing of options (which would make your test to have to be updated). The reason is that users embedding the server would not benefit from that if they configure the server directly with options.
I recommend doing checks/validations in code that is done in NewServer() which then means it applies to both config and direct options. For leafnode, this would be `validateLeafNode()`.

Reply:
Aaah, ok. Great thanks, will take another look.